From b4d636ca126ee497c280eb04ea3aee0214b481ac Mon Sep 17 00:00:00 2001
From: David Woodhouse <David.Woodhouse@intel.com>
Date: Mon, 8 Dec 2014 14:08:20 +0000
Subject: [PATCH 0132/1800] Bug 741059 - Prompts for login PIN with
 CKF_PROTECTED_AUTHENTICATION_PATH

This turns out to be relatively simple to fix. Based on the patch in
http://bugzilla.mozilla.org/show_bug.cgi?id=229023
---
 smime/lib/e-cert-db.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/smime/lib/e-cert-db.c b/smime/lib/e-cert-db.c
index 3aeb04763a..1c702ac1f8 100644
--- a/smime/lib/e-cert-db.c
+++ b/smime/lib/e-cert-db.c
@@ -453,6 +453,11 @@ pk11_password (PK11SlotInfo *slot,
 
 	gboolean rv = FALSE;
 
+	/* For tokens with CKF_PROTECTED_AUTHENTICATION_PATH we
+	 * need to return a non-empty but unused password */
+	if (PK11_ProtectedAuthenticationPath(slot))
+		return PORT_Strdup("");
+
 	g_signal_emit (
 		e_cert_db_peek (),
 		e_cert_db_signals[PK11_PASSWD], 0,
-- 
2.15.0

